packages/components/src/components/radio-group/radio.jsx

```diff
@@ -2,74 +2,56 @@ import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import React from 'react';
 
-class Radio extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            checked: props.defaultChecked,
-        };
-    }
+const Radio = ({
+    children,
+    className,
+    classNameLabel,
+    defaultChecked,
+    id,
+    onChange, // This needs to be here so it's not included in `otherProps`
+    ...otherProps
+}) => {
+    const [checked, setChecked] = React.useState(defaultChecked);
 
     /*
-     * We use componentDidUpdate here to tell the Radio component to update itself
+     * We use useEffect here to tell the Radio component to update itself
      * when it's no longer selected
      * This is because we're handling the state of what's selected in RadioGroup with the defaultChecked prop
      */
-    componentDidUpdate(prevProps) {
-        if (this.props.defaultChecked === prevProps.defaultChecked) {
-            return;
-        }
-
-        this.setState({ checked: this.props.defaultChecked });
-    }
+    React.useEffect(() => {
+        setChecked(defaultChecked);
+    }, [defaultChecked]);
 
-    onChange = e => {
-        this.setState({ checked: e.target.checked });
-        this.props.onChange(e);
+    const onInputChange = e => {
+        setChecked(e.target.checked);
+        onChange(e);
     };
 
-    render() {
-        const {
-            className,
-            classNameLabel,
-            id,
-            children,
-            onChange, // This needs to be here so it's not included in `otherProps`
-            ...otherProps
-        } = this.props;
-
-        return (
-            <label
-                htmlFor={id}
-                className={classNames('dc-radio-group__item', className, {
-                    'dc-radio-group__item--active': this.state.checked,
+    return (
+        <label
+            htmlFor={id}
+            className={classNames('dc-radio-group__item', className, {
+                'dc-radio-group__item--active': checked,
+            })}
+        >
+            <input className='dc-radio-group__input' type='radio' id={id} onChange={onInputChange} {...otherProps} />
```

Comment:
If the input is hidden, we should put the id on the element that accepts the mouse events.

Reply:
I don't think we may have a case where `input` is hidden but its label (whole wrapper) isn't. So in order for a radio to be hidden we need to pass a dedicated `is_hidden` prop to cover all the `jsx` here.